src/core/probe/manager.rs

```diff
@@ -144,6 +148,19 @@ impl ProbeManager {
         Ok(())
     }
 
+    /// Request a filter to be attached to all probes.
+    ///
+    /// ```
+    /// mgr.register_filter(filter)?;
+    /// ```
+    pub(crate) fn register_filter(&mut self, key: String, filter: Filter) -> Result<()> {
```

Comment:
Initially, it was a vector and the idea was to make available this kind of filtering (Packets) to USDTs as this may technically work with them (although not yet verified).
For that, the command line options should be considered, as they should be consistent.
The truth is that it's probably very unlikely we'll end up using this with USDTs, but that said, there's no way for the user ATM to include two filters of this same type into the vector. The simple approach was then, let's use vector and expand later if we have a use case.